xchange-stream-coinmate/src/main/java/info/bitrich/xchangestream/coinmate/v2/CoinmateStreamingService.java

```diff
@@ -0,0 +1,88 @@
+package info.bitrich.xchangestream.coinmate.v2;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import info.bitrich.xchangestream.coinmate.v2.dto.CoinmateAuthenticedSubscribeMessage;
+import info.bitrich.xchangestream.coinmate.v2.dto.CoinmatePingMessage;
+import info.bitrich.xchangestream.coinmate.v2.dto.CoinmateSubscribeMessage;
+import info.bitrich.xchangestream.coinmate.v2.dto.CoinmateUnsubscribeMessage;
+import info.bitrich.xchangestream.coinmate.v2.dto.auth.AuthParams;
+import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.websocketx.TextWebSocketFrame;
+
+import java.io.IOException;
+
+/**
+ * Coinmate Websocket API 2.0
+ */
+class CoinmateStreamingService extends JsonNettyStreamingService {
+
+  private final AuthParams authParams;
+
+  CoinmateStreamingService(String url, AuthParams authParams) {
+    super(url);
+    this.authParams = authParams;
+  }
+
+  @Override
+  protected String getChannelNameFromMessage(JsonNode message) throws IOException {
+    String event = message.get("event").asText();
+    if (!event.equals("data")) {
+      return null;
+    }
+    return message.get("channel").asText();
+  }
+
+  @Override
+  public String getSubscribeMessage(String channelName, Object... args) throws IOException {
+    if (args.length > 0 && args[0].equals(true)) {
+      // needs authentication
+      CoinmateAuthenticedSubscribeMessage subscribeMessage =
+          generateAuthenticatedSubscribeMessage(channelName);
+      return objectMapper.writeValueAsString(subscribeMessage);
```

Comment:
can you just inline this here and below as opposed to creating variables

Reply:
Please, what do you mean by inlining? Constructing the JSON manually as a String?

Reply:
```
      return objectMapper.writeValueAsString(
          generateAuthenticatedSubscribeMessage(channelName));
```

Reply:
thanks, fixed